src/Kmd.Logic.Gateway.Automation/Gateway/GatewayDetails.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Kmd.Logic.Gateway.Automation.Gateway
+{
+    public class GatewayDetails
+    {
+#pragma warning disable CA2227 // Collection properties should be read only
+        public IList<Product> Products { get; set; }
```

Comment:
Could we made it IReadOnlyList or `IList` with only getter.

Reply:
IReadOnlyList or IEnumerable is ok.

Reply:
yaml deserialization will throw exception if it is readonly